tests/queries/0_stateless/02494_query_cache_use_database.sql

```diff
@@ -0,0 +1,30 @@
+-- Tags: no-parallel, no-fasttest
+-- Tag no-fasttest: Depends on OpenSSL
+-- Tag no-parallel: Messes with internal cache
+
+-- Test for issue #64136
+
+SYSTEM DROP QUERY CACHE;
+
+DROP DATABASE IF EXISTS db1;
+DROP DATABASE IF EXISTS db2;
+
+CREATE DATABASE db1;
+CREATE DATABASE db2;
+
+CREATE TABLE db1.tab(a UInt64, PRIMARY KEY a);
+CREATE TABLE db2.tab(a UInt64, PRIMARY KEY a);
+
+INSERT INTO db1.tab values(1);
+INSERT INTO db2.tab values(2);
+
+USE db1;
+SELECT * FROM tab SETTINGS use_query_cache=1;
+
+USE db2;
+SELECT * FROM tab SETTINGS use_query_cache=1;
```

Comment:
would this work for cases where the database is set via the connection string ? snippet from java client
```
ClickHouseNode.builder()
			.host(host)
			.port(ClickHouseProtocol.HTTP, port)
			.options(Map.of(ClickHouseClientOption.SSL.getKey(), "true"))
			.database(db) <-- this
			.credentials(credentials)
			.build();
```

Reply:
I don't have Java locally set up, but I would assume it would work.

It would be cool if you could check what

```sql
SELECT currentDatabase();
```

returns after running your snippet.

Reply:
java does this HTTP request

```
curl 'localhost:8123/?database=test&query=select+currentDatabase()&use_query_cache=1&query_cache_nondeterministic_function_handling=save'
test

curl 'localhost:8123/?database=default&query=select+currentDatabase()&use_query_cache=1&query_cache_nondeterministic_function_handling=save'
test
```

Reply:
> I don't have Java locally set up, but I would assume it would work.
> 
> It would be cool if you could check what
> 
> ```sql
> SELECT currentDatabase();
> ```
> 
> returns after running your snippet.

yep returns the selected database from the query. lgtm